### PR TITLE
Implementation of TUA229, TUA231, TUA359 and changes in pages (xPath upd, added methods)

### PR DESCRIPTION
**Pull request by janestadnytska**
- Implementation of TUA229, TUA231, TUA359
 - added waitForElementToDisappear into Base class
 - corrected xPath in AddChallengePage
 - added getBorderColor methods into EditProfileComponent & AddChallengePage
 - added getProfilenode method into ProfilePage class

### CODE REVIEW

**Review by Maksym637 — APPROVED** [review]

File: src/main/java/org/ssu/edu/teachua/ui/pages/challenges/AddChallengePage.java [hunk 1]
```diff
@@ -169,4 +169,20 @@ public ViewChallengePage clickViewChallenge() {
...
+        return borderColor;
+    }
+    public String getBorderColorForTitleField() {
+        String borderColor= title.getCssValue("border-color");
+        return borderColor;
+    }
+    public String getBorderColorForDescriptionField() {
+        String borderColor= description.getCssValue("border-color");
+        return borderColor;
+    }
```
Comment (Maksym637) [inline src/main/java/org/ssu/edu/teachua/ui/pages/challenges/AddChallengePage.java:187]: Variables are redundant

File: src/main/java/org/ssu/edu/teachua/ui/components/modal/EditProfileComponent.java [hunk 2]
```diff
@@ -149,7 +158,10 @@ public EditProfileComponent confirmNewPassword(String password) {
     public String getAlertMessageConfirmPassword() {
         return waitForElementToAppear(alertMessageConfirmPassword).getText();
     }
-
+    public String getBorderColorForConfirmPasswordField() {
+        String borderColor= confirmPasswordField.getCssValue("border-color");
+        return borderColor;
+    }
```
Comment (Maksym637) [inline src/main/java/org/ssu/edu/teachua/ui/components/modal/EditProfileComponent.java:164]: Variable are redundant

File: src/main/java/org/ssu/edu/teachua/ui/pages/challenges/AddChallengePage.java (hunk 1 above)
Comment (janestadnytska) [reply src/main/java/org/ssu/edu/teachua/ui/pages/challenges/AddChallengePage.java]: fixed

File: src/test/java/org/ssu/edu/teachua/ui/TUA229.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    public void testTUA229() throws IOException {
+
+        TestValueProvider adminCredentials = new TestValueProvider();
+
+        AddChallengePage addChallengePage = new HomePage(driver)
+                .getHeader()
+                .openGuestProfileMenu()
+                .openLogInForm()
+                .enterEmail(adminCredentials.getAdminEmail())
+                .enterPassword(adminCredentials.getAdminPassword())
```
Comment (lhalam) [inline src/test/java/org/ssu/edu/teachua/ui/TUA229.java:23]: valueProvider.getAdminEmail()

File: src/test/java/org/ssu/edu/teachua/ui/TUA229.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+
+public class TUA229 extends TestRunnerUI {
+    @Test
+    public void testTUA229() throws IOException {
+
+        TestValueProvider adminCredentials = new TestValueProvider();
```
Comment (lhalam) [inline src/test/java/org/ssu/edu/teachua/ui/TUA229.java:16]: remove this

File: src/test/java/org/ssu/edu/teachua/ui/TUA229.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                .enterEmail(adminCredentials.getAdminEmail())
+                .enterPassword(adminCredentials.getAdminPassword())
+                .clickLoginButton()
+                .getHeader()
+                .openAdminProfileMenu()
+                .openContentMenu()
+                .openChallengesMenu()
+                .clickChallenges()
+                .addChallenge()
+                .addPhoto("C:/test/image.png");
```
Comment (lhalam) [inline src/test/java/org/ssu/edu/teachua/ui/TUA229.java:31]: change this to use resources folder

**Review by lhalam — APPROVED** [review]

**Pull request merged**